packages/collector/Collector_Content.php

```diff
@@ -0,0 +1,18 @@
+<?php
+function collector_iterate_directory($path, $prefix, $callback)
+{
+	$handle = opendir($path);
+
+	while($entry = readdir($handle))
+	{
+		if($entry === '.' || $entry === '..' || substr($entry, 0, 1) === '.')
+		{
+			continue;
+		}
+
+		$realPath = realpath($path . '/' . $entry);
```

Comment:
would https://developer.wordpress.org/reference/functions/path_join/ be a better fit here?